CopilotKit/packages/react-textarea/src/components/hovering-toolbar/hovering-toolbar.tsx

```diff
@@ -51,57 +53,82 @@ export const HoveringToolbar = (props: HoveringToolbarProps) => {
 
     const domRange = domSelection.getRangeAt(0);
     const rect = domRange.getBoundingClientRect();
-
+    
     // We use window = (0,0,0,0) as a signal that the selection is not in the original copilot-textarea,
     // but inside the hovering window.
     //
     // in such case, we simply do nothing.
     if (rect.top === 0 && rect.left === 0 && rect.width === 0 && rect.height === 0) {
       return;
     }
-    const minGapFromEdge = 60;
-    const verticalOffsetFromCorner = 35;
-    const horizontalOffsetFromCorner = 15;
-    let top = rect.top + window.scrollY - el.offsetHeight + verticalOffsetFromCorner;
-    // make sure top is in the viewport and not too close to the edge
-    if (top < minGapFromEdge) {
-      top = rect.bottom + window.scrollY + minGapFromEdge;
```

Comment:
This case is possible only in a crazy scenario when the textarea is positioned above the start of the document. I don't think it's worth caring about this case. Fix me if I'm wrong and I rewrite it like `let top = Math.max(verticalOffsetFromCorner, rect.bottom + window.scrollY + verticalOffsetFromCorner);`